webpack/JobInvocationDetail/JobInvocationSystemStatusChart.js

```diff
@@ -0,0 +1,143 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { translate as __ } from 'foremanReact/common/I18n';
+import {
+  ChartDonut,
+  ChartLabel,
+  ChartLegend,
+  ChartTooltip,
+} from '@patternfly/react-charts';
+import {
+  DescriptionList,
+  DescriptionListTerm,
+  DescriptionListGroup,
+  DescriptionListDescription,
+  FlexItem,
+  Text,
+} from '@patternfly/react-core';
+import {
+  global_palette_green_500 as successedColor,
+  global_palette_red_100 as failedColor,
+  global_palette_blue_300 as inProgressColor,
+  global_palette_black_600 as canceledColor,
+} from '@patternfly/react-tokens';
+import DefaultLoaderEmptyState from 'foremanReact/components/HostDetails/DetailsCard/DefaultLoaderEmptyState';
+import './JobInvocationDetail.scss';
+
+const JobInvocationSystemStatusChart = ({
+  data,
+  isAlreadyStarted,
+  formattedStartDate,
+}) => {
+  const {
+    succeeded,
+    failed,
+    pending,
+    cancelled,
+    total,
+    total_hosts: totalHosts, // includes scheduled
+  } = data;
+  const chartData = [
+    { title: __('Succeeded:'), count: succeeded, color: successedColor.value },
+    { title: __('Failed:'), count: failed, color: failedColor.value },
+    { title: __('In Progress:'), count: pending, color: inProgressColor.value },
+    { title: __('Canceled:'), count: cancelled, color: canceledColor.value },
+  ];
+  const chartDonutTitle = () => {
+    if (total > 0) return `${succeeded.toString()}/${total}`;
+    if (totalHosts > 0) return `0/${totalHosts}`;
+    return '0';
+  };
+
+  const [legendWidth, setLegendWidth] = useState(270);
+
+  useEffect(() => {
+    const legendContainer = document.querySelector('.chart-legend');
+    if (legendContainer) {
+      const rectElement = legendContainer.querySelector('rect');
+      if (rectElement) {
+        const rectWidth = parseFloat(rectElement.getAttribute('width'));
+        setLegendWidth(rectWidth);
+      }
+    }
+  }, [isAlreadyStarted, data]);
+
+  return (
+    <>
+      <FlexItem className="chart-donut">
+        <ChartDonut
+          allowTooltip
+          constrainToVisibleArea
+          data={
+            total > 0
+              ? chartData.map(d => ({
+                  label: `${d.title} ${d.count} hosts`,
+                  y: d.count,
+                }))
+              : [{ label: `Scheduled: ${totalHosts} hosts`, y: 1 }]
+          }
+          colorScale={total > 0 ? chartData.map(d => d.color) : ['#8A8D90']}
+          labelComponent={
+            <ChartTooltip pointerLength={0} constrainToVisibleArea />
+          }
+          title={chartDonutTitle}
+          titleComponent={<ChartLabel style={{ fontSize: '20px' }} />}
+          subTitle={__('Systems')}
+          subTitleComponent={
+            <ChartLabel
+              style={{ fontSize: '12px', fill: canceledColor.value }}
```

Comment:
Should be in a css file

Reply:
> Should be in a css file

@MariaAga is the comment "// inline style overrides PatternFly default styling" enough?

Reply:
yeah! i approved it, and have to merge permissions in plugins, only @adamruzicka can merge here:)

Reply:
I didn't want to merge with outstanding comments in case this was still unresolved